src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -224,37 +254,43 @@ def _create_users_tasks(self) -> List[asyncio.Task[None]]:
             )
         return tasks
 
-    def _get_upper_bound(self, batteries: Set[int]) -> float:
+    def _get_upper_bound(self, batteries: Set[int], must_use_all: bool) -> float:
         """Get total upper bound of power to be set for given batteries.
 
         Note, output of that function doesn't guarantee that this bound will be
         the same when the request is processed.
 
         Args:
             batteries: List of batteries
+            must_use_all: whether all batteries in the power request must be used.
```

Comment:
> Also the documentation should clarify what "all" means, is this all the batteries in the batteries set? 

It means all the batteries included in the `batteries` set in the request. I'll amend it but i thought it was clear given the context.

>  Also explicitly mention which batteries are normally used when include_broken is either True or False (like batteries in state X and Y are not used normally but if include_broken is True they will be used too).

This might be better added once at the entry point when the component data is retrieved